packages/netlify/src/ssr-function.ts

```diff
@@ -8,15 +8,15 @@ applyPolyfills();
 // biome-ignore lint/complexity/noBannedTypes: safe to use in this case
 export type Args = {};
 
-const clientAddressSymbol = Symbol.for('astro.clientAddress');
-
 export const createExports = (manifest: SSRManifest, _args: Args) => {
 	const app = new App(manifest);
 
-	function createHandler(integrationConfig: { cacheOnDemandPages: boolean }) {
+	function createHandler(integrationConfig: { cacheOnDemandPages: boolean, notFoundContent?: string }) {
```

Comment:
So it seems like the approach is to make the serverless function serve the static file by inlining 404.html into the code.

How feasible would it be to make the serverless function act only on certain routes?

This is how it's done in the vercel adapter: https://github.com/withastro/astro/pull/9648/files#diff-071ccfbdfd7b9bc803219179f88ae0bdb227a399bc4a481b7831c82b801d7a17R290-R296

Reply:
This would address the (admittedly small) billing concern.

Reply:
That would require us to generate a `_redirects` file, which opens up some complexity that i'd prefer not to have. The single benefit from that, as I understand it, would be that there's no function execution for 404 pages. I'm unsure how valuable that really is, and if we do that change i'd prefer to do it in a separate PR in the future.